### PR TITLE
Update path to download pkgconf source from and bump version

### LINKED ISSUE #18304: [package] pkgconf/1.9.3: Cannot build from source - SSL cert for https://distfiles.ariadne.space expired
### Description

When I attempt to build `pkgconf` from source I get a failure with the following error:
```
pkgconf/1.9.3: Calling source() in /Users/dheater/.conan2/p/pkgco8f17a0a1eb9e1/s/src
pkgconf/1.9.3: ERROR: Error downloading file https://distfiles.dereferenced.org/pkgconf/pkgconf-1.9.3.tar.xz: 'HTTPSConnectionPool(host='distfiles.dereferenced.org', port=443): Max retries exceeded with url: /pkgconf/pkgconf-1.9.3.tar.xz (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x111f49cc0>, 'Connection to distfiles.dereferenced.org timed out. (connect timeout=30)'))'
pkgconf/1.9.3: Waiting 5 seconds to retry...
```

When I check https://distfiles.ariadne.space, its cert is clearly expired.

It looks like `pkgconf` binaries are hosted on GitHub now at https://github.com/pkgconf/pkgconf so maybe the package should be updated to reference GiitHub

### Package and Environment Details

* Package Name/Version: **pkgconf/1.9.3**
* Operating System+version: **MacOS Ventura**
* Compiler+version: **GCC 8**
* Docker image: **conanio/gcc8**
* Conan version: **conan 1.18.0**
* Python version: **Python 3.7.4**

### Conan profile

[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=14
os=Macos

### Steps to reproduce

conan install --requires=pkgconf/1.9.3 --build="*"

### Logs

<details><summary>Click to expand log</summary>

```
ERROR: Exiting with code: 3
❯ conan remove pkgconf
Remove the recipe and all the packages of 'pkgconf/1.9.3#a920b5c7f8d04f22b9fe03db91a864dd'? (yes/no): yes
Remove summary:
Local Cache
  pkgconf/1.9.3#a920b5c7f8d04f22b9fe03db91a864dd: Removed recipe and all binaries
❯ conan install --requires=pkgconf/1.9.3 --build="*"

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=14
os=Macos

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=14
os=Macos


======== Computing dependency graph ========
pkgconf/1.9.3: Not found in local cache, looking in remotes...
pkgconf/1.9.3: Checking remote: sl-artifactory
pkgconf/1.9.3: Downloaded recipe revision a920b5c7f8d04f22b9fe03db91a864dd
Graph root
    cli
Requirements
    pkgconf/1.9.3#a920b5c7f8d04f22b9fe03db91a864dd - Downloaded (sl-artifactory)
Build requirements
    meson/1.0.0#15586c0ac6f682805875ef903dbe7ee2 - Cache
    ninja/1.11.1#a2f0b832705907016f336839f96963f8 - Cache

======== Computing necessary packages ========
ninja/1.11.1: Forced build from source
meson/1.0.0: Forced build from source
pkgconf/1.9.3: Forced build from source
Requirements
    pkgconf/1.9.3#a920b5c7f8d04f22b9fe03db91a864dd:8b5cf1367c351fd3c60ef4e28c1a50612761b13e - Build
Build requirements
    meson/1.0.0#15586c0ac6f682805875ef903dbe7ee2:da39a3ee5e6b4b0d3255bfef95601890afd80709 - Build
    ninja/1.11.1#a2f0b832705907016f336839f96963f8:9ac8640923e5284645f8852ef8ba335654f4020e - Build

======== Installing packages ========

-------- Installing package ninja/1.11.1 (1 of 3) --------
ninja/1.11.1: Building from source
ninja/1.11.1: Package ninja/1.11.1:9ac8640923e5284645f8852ef8ba335654f4020e
ninja/1.11.1: Copying sources to build folder
ninja/1.11.1: Building your package in /Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/b
ninja/1.11.1: Calling generate()
ninja/1.11.1: Generators folder: /Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/b/build/Release/generators
ninja/1.11.1: CMakeToolchain generated: conan_toolchain.cmake
ninja/1.11.1: CMakeToolchain generated: CMakePresets.json
ninja/1.11.1: CMakeToolchain generated: ../../../src/CMakeUserPresets.json
ninja/1.11.1: Generating aggregated env files
ninja/1.11.1: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
ninja/1.11.1: Calling build()
ninja/1.11.1: Running CMake.configure()
ninja/1.11.1: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/dheater/.conan2/p/b/ninjaee73fbcdbf54
-- Using Conan toolchain: /Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/b/build/Release/generators/conan_toolchai
-- Conan toolchain: C++ Standard 17 with extensions OFF
-- The C compiler identification is AppleClang 14.0.3.14030022
-- The CXX compiler identification is AppleClang 14.0.3.14030022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- IPO / LTO enabled
-- Performing Test flag_no_deprecated
-- Performing Test flag_no_deprecated - Success
-- Performing Test flag_color_diag
-- Performing Test flag_color_diag - Success
CMake Warning at CMakeLists.txt:49 (message):
  re2c was not found; changes to src/*.in.cc will not affect your build.


-- Looking for fork
-- Looking for fork - found
-- Looking for pipe
-- Looking for pipe - found
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: /Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/b/build/Release

ninja/1.11.1: Running CMake.build()
ninja/1.11.1: RUN: cmake --build "/Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/b/build/Release" -- -j16
[  3%] Building CXX object CMakeFiles/libninja-re2c.dir/src/depfile_parser.cc.o
[  6%] Building CXX object CMakeFiles/libninja-re2c.dir/src/lexer.cc.o
[  9%] Building CXX object CMakeFiles/libninja.dir/src/build_log.cc.o
[ 12%] Building CXX object CMakeFiles/libninja.dir/src/clean.cc.o
[ 16%] Building CXX object CMakeFiles/libninja.dir/src/dyndep.cc.o
[ 19%] Building CXX object CMakeFiles/libninja.dir/src/build.cc.o
[ 22%] Building CXX object CMakeFiles/libninja.dir/src/disk_interface.cc.o
[ 25%] Building CXX object CMakeFiles/libninja.dir/src/clparser.cc.o
[ 29%] Building CXX object CMakeFiles/libninja.dir/src/debug_flags.cc.o
[ 32%] Building CXX object CMakeFiles/libninja.dir/src/dyndep_parser.cc.o
[ 35%] Building CXX object CMakeFiles/libninja.dir/src/edit_distance.cc.o
[ 38%] Building CXX object CMakeFiles/libninja.dir/src/eval_env.cc.o
[ 41%] Building CXX object CMakeFiles/libninja.dir/src/graph.cc.o
[ 45%] Building CXX object CMakeFiles/libninja.dir/src/deps_log.cc.o
[ 48%] Building CXX object CMakeFiles/libninja.dir/src/graphviz.cc.o
[ 51%] Building CXX object CMakeFiles/libninja.dir/src/json.cc.o
[ 54%] Building CXX object CMakeFiles/libninja.dir/src/line_printer.cc.o
[ 58%] Building CXX object CMakeFiles/libninja.dir/src/manifest_parser.cc.o
[ 61%] Building CXX object CMakeFiles/libninja.dir/src/metrics.cc.o
[ 64%] Building CXX object CMakeFiles/libninja.dir/src/missing_deps.cc.o
[ 67%] Building CXX object CMakeFiles/libninja.dir/src/parser.cc.o
[ 70%] Building CXX object CMakeFiles/libninja.dir/src/state.cc.o
[ 70%] Built target libninja-re2c
[ 74%] Building CXX object CMakeFiles/libninja.dir/src/status.cc.o
[ 77%] Building CXX object CMakeFiles/libninja.dir/src/string_piece_util.cc.o
[ 80%] Building CXX object CMakeFiles/libninja.dir/src/util.cc.o
[ 83%] Building CXX object CMakeFiles/libninja.dir/src/version.cc.o
[ 87%] Building CXX object CMakeFiles/libninja.dir/src/subprocess-posix.cc.o
[ 87%] Built target libninja
[ 90%] Generating build/browse_py.h
[ 93%] Building CXX object CMakeFiles/ninja.dir/src/ninja.cc.o
[ 96%] Building CXX object CMakeFiles/ninja.dir/src/browse.cc.o
[100%] Linking CXX executable ninja
[100%] Built target ninja

ninja/1.11.1: Package '9ac8640923e5284645f8852ef8ba335654f4020e' built
ninja/1.11.1: Build folder /Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/b/build/Release
ninja/1.11.1: Generating the package
ninja/1.11.1: Temporary package folder /Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/p
ninja/1.11.1: Calling package()
ninja/1.11.1: Running CMake.install()
ninja/1.11.1: RUN: cmake --install "/Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/b/build/Release" --prefix "/Use
-- Install configuration: "Release"
-- Installing: /Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/p/bin/ninja

ninja/1.11.1: package(): Packaged 2 files: ninja, COPYING
ninja/1.11.1: Created package revision ff840b94c3026b3fdb1ab1418f7822a3
ninja/1.11.1: Package '9ac8640923e5284645f8852ef8ba335654f4020e' created
ninja/1.11.1: Full package reference: ninja/1.11.1#a2f0b832705907016f336839f96963f8:9ac8640923e5284645f8852ef8ba3
ninja/1.11.1: Package folder /Users/dheater/.conan2/p/b/ninjaee73fbcdbf544/p

-------- Installing package meson/1.0.0 (2 of 3) --------
meson/1.0.0: Building from source
meson/1.0.0: Package meson/1.0.0:da39a3ee5e6b4b0d3255bfef95601890afd80709
meson/1.0.0: Building your package in /Users/dheater/.conan2/p/b/mesonc056e05094bc0/b
meson/1.0.0: Generating aggregated env files
meson/1.0.0: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
meson/1.0.0: Calling build()
meson/1.0.0: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' built
meson/1.0.0: Build folder /Users/dheater/.conan2/p/b/mesonc056e05094bc0/b/build
meson/1.0.0: Generating the package
meson/1.0.0: Temporary package folder /Users/dheater/.conan2/p/b/mesonc056e05094bc0/p
meson/1.0.0: Calling package()
meson/1.0.0: package(): Packaged 8 files
meson/1.0.0: package(): Packaged 251 '.py' files
meson/1.0.0: package(): Packaged 1 '.toml' file: pyproject.toml
meson/1.0.0: package(): Packaged 2 '.in' files: MANIFEST.in, .coveragerc.in
meson/1.0.0: package(): Packaged 2 '.md' files: README.md, contributing.md
meson/1.0.0: package(): Packaged 1 '.cfg' file: setup.cfg
meson/1.0.0: package(): Packaged 1 '.cmd' file: meson.cmd
meson/1.0.0: package(): Packaged 2 '.1' files: meson.1, foo.1
meson/1.0.0: package(): Packaged 1 '.rtf' file: License.rtf
meson/1.0.0: package(): Packaged 3 '.html' files: conclusion.html, license.html, welcome.html
meson/1.0.0: package(): Packaged 27 '.txt' files
meson/1.0.0: package(): Packaged 1 '.ui' file: mywidget.ui
meson/1.0.0: package(): Packaged 12 '.build' files
meson/1.0.0: package(): Packaged 1 '.xml' file: my-resources.xml
meson/1.0.0: package(): Packaged 1 '.vala' file: mywidget.vala
meson/1.0.0: package(): Packaged 10 '.c' files
meson/1.0.0: package(): Packaged 9 '.wrap' files
meson/1.0.0: package(): Packaged 1 '.h' file: lib.h
meson/1.0.0: package(): Packaged 5 '.sh' files
meson/1.0.0: package(): Packaged 1 '.cpp' file: myapp.cpp
meson/1.0.0: package(): Packaged 1 '.icns' file: myapp.icns
meson/1.0.0: package(): Packaged 1 '.gz' file: template.dmg.gz
meson/1.0.0: package(): Packaged 1 '.iss' file: myapp.iss
meson/1.0.0: package(): Packaged 1 '.plist' file: Info.plist
meson/1.0.0: package(): Packaged 1 '.dat' file: foo.dat
meson/1.0.0: package(): Packaged 4 '.json' files: linux-mingw-w64-32bit.json, linux-mingw-w64-64bit.json, ubuntu-armhf.json, test.schema.json
meson/1.0.0: package(): Packaged 1 '.cmake' file: preload.cmake
meson/1.0.0: package(): Packaged 1 '.ps1' file: cmd_or_ps.ps1
meson/1.0.0: package(): Packaged 1 '.png' file: meson_logo_big.png
meson/1.0.0: package(): Packaged 2 '.svg' files: wrap_logo.svg, meson_logo.svg
meson/1.0.0: package(): Packaged 1 '.xsd' file: schema.xsd
meson/1.0.0: package(): Packaged 1 '.policy' file: com.mesonbuild.install.policy
meson/1.0.0: package(): Packaged 1 '.meson' file: macros.meson
meson/1.0.0: package(): Packaged 4 '.vim' files: meson.vim, meson.vim, meson.vim, meson.vim
meson/1.0.0: Created package revision 5c8fd51fc33f12e26519674d99afd0e5
meson/1.0.0: Package 'da39a3ee5e6b4b0d3255bfef95601890afd80709' created
meson/1.0.0: Full package reference: meson/1.0.0#15586c0ac6f682805875ef903dbe7ee2:da39a3ee5e6b4b0d3255bfef95601890afd80709#5c8fd51fc33f12e26519674d99afd0e5
meson/1.0.0: Package folder /Users/dheater/.conan2/p/b/mesonc056e05094bc0/p
pkgconf/1.9.3: Sources downloaded from 'sl-artifactory'
pkgconf/1.9.3: Calling source() in /Users/dheater/.conan2/p/pkgco8f17a0a1eb9e1/s/src
pkgconf/1.9.3: ERROR: Error downloading file https://distfiles.dereferenced.org/pkgconf/pkgconf-1.9.3.tar.xz: 'HTTPSConnectionPool(host='distfiles.dereferenced.org', port=443): Max retries exceeded with url: /pkgconf/pkgconf-1.9.3.tar.xz (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x1124423b0>, 'Connection to distfiles.dereferenced.org timed out. (connect timeout=30)'))'
pkgconf/1.9.3: Waiting 5 seconds to retry...
pkgconf/1.9.3: ERROR: Error downloading file https://distfiles.dereferenced.org/pkgconf/pkgconf-1.9.3.tar.xz: 'HTTPSConnectionPool(host='distfiles.dereferenced.org', port=443): Max retries exceeded with url: /pkgconf/pkgconf-1.9.3.tar.xz (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x112442620>, 'Connection to distfiles.dereferenced.org timed out. (connect timeout=30)'))'
pkgconf/1.9.3: Waiting 5 seconds to retry...
ERROR: pkgconf/1.9.3: Error in source() method, line 59
        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
        ConanException: Error downloading file https://distfiles.dereferenced.org/pkgconf/pkgconf-1.9.3.tar.xz: 'HTTPSConnectionPool(host='distfiles.dereferenced.org', port=443): Max retries exceeded with url: /pkgconf/pkgconf-1.9.3.tar.xz (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x1124429e0>, 'Connection to distfiles.dereferenced.org timed out. (connect timeout=30)'))'
```

</details>

### PR DESCRIPTION
**Pull request by dheater**
Specify library name and version:  **pkgconf/1.9.5**

The `pkgconf` maintainers had to change the host domain for their source release. See https://github.com/pkgconf/pkgconf/issues/302 for details. Updating to the new path here.

Also adding version 1.9.5

Closes  #18304
---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/pkgconf/all/patches/1.9.5-0001-PKG_CONF_PATH-allow-colon+semicolon-separator.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
++++ libpkgconf/path.c
+@@ -138,7 +138,7 @@
+ 		return 0;
+ 
+ 	iter = workbuf = strdup(text);
+-	while ((p = strtok(iter, PKG_CONFIG_PATH_SEP_S)) != NULL)
++	while ((p = strtok(iter, ":;")) != NULL)
+ 	{
+ 		pkgconf_path_add(p, dirlist, filter);
+
```
Comment (ericLemanissier) [inline recipes/pkgconf/all/patches/1.9.5-0001-PKG_CONF_PATH-allow-colon+semicolon-separator.patch:11]:
```suggestion
```

**Review by ericLemanissier — APPROVED** [review]
Please re-use the patch if it's the same

**Review by jwillikers — APPROVED** [review]

**Comment by dheater** [comment]
Not getting an email or report here on why *continuous-integration/jenkins/pr-merge* is failing. Details link is 403.

**Comment by ericLemanissier** [comment]
please revert the last commit. patches were ok, cf https://c3i.jfrog.io/c3i/misc-v2/logs/pr/18307/5-windows-msvc/pkgconf/1.9.5/43771b8671ac44479c188dd72670e2eb2d7918a6-build.txt. CI sometimes fails for obscure reasons. If you don't get any message after a few hours, you can close and reopen the PR to retrigger a build

**Comment by ericLemanissier** [comment]
I was probably not clear enough in my last message. Please simply revert 575cd1653697b755dd7d85f55e2f8bc0e07ece44

**Review by ericLemanissier — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by memsharded — COMMENTED** [review]
Many thanks for your contribution!

**Review by memsharded — APPROVED** [review]
Many thanks for your contribution!

**Pull request merged**